### PR TITLE
 bpfman: Make TC and XDP dispatcher configurable

### PR DESCRIPTION
**Pull request by ShockleyJE**
Requested in upstream issue [1120](https://github.com/bpfman/bpfman/issues/1220)

### CODE REVIEW

**Comment by ShockleyJE** [comment]
Overlooked the need to replace the actual accessing of the const `XDP_DISPATCHER_IMAGE` and `TC_DISPATCHER_IMAGE` values  with access via the config object. 

I considered moving these constants out of `lib.rs` to prevent future direct access accidents, but was unsure whether any CI/CD touched these values due to the comment block:

```rust
    // When release is made, the dispatcher images should be changed to the
    // release tag, for example "v0.5.4". After release is made, the tag should
    // be changed back to the "latest" release tag. Note also that these values should
    // be read through the public interfaces provided by config::Config's RegistryConfig
```

**Comment by anfredette** [comment]
Thanks for your contribution, James!

File: bpfman/src/config.rs [hunk 1]
```diff
@@ -36,6 +38,15 @@
     pub(crate) fn database(&self) -> &Option<DatabaseConfig> {
         &self.database
     }
+
+    pub(crate) fn set_registry(&mut self, registry: RegistryConfig) {
```
[the remark this answers is not in the document]
Comment (Billy99) [reply bpfman/src/config.rs:42]: You can delete this function. Not used.

**Review by Billy99 — COMMENTED** [review]
Thanks for the changes. Only a few requests:
* Update [scripts/bpfman.toml](https://github.com/bpfman/bpfman/blob/main/scripts/bpfman.toml) with your new fields. I just put the default values in the file, but it helps the user see how to format the values.
* Update the docs with your new fields [docs/developer-guide/configuration.md](https://github.com/bpfman/bpfman/blob/main/docs/developer-guide/configuration.md)
* Need to run fmt and clippy against the PR (we currently have nightly pinned):
`cargo +nightly-2024-09-24 fmt --all -- --check`
`cargo +nightly-2024-09-24 clippy --all-targets --all-features -- --deny warnings`

**Review by Billy99 — APPROVED** [review]
/lgtm

Thanks!

**Pull request merged**